pkg/cmd/cli/cmd/volume.go

```diff
@@ -590,14 +678,68 @@ func (v *VolumeOptions) removeVolumeFromSpec(spec *kapi.PodSpec, info *resource.
 	return nil
 }
 
+func sourceAccessMode(readOnly bool) string {
+	if readOnly {
+		return " read-only"
+	}
+	return ""
+}
+
+func describePersistentVolumeClaim(claim *kapi.PersistentVolumeClaim) string {
+	if len(claim.Spec.VolumeName) == 0 {
+		// TODO: check for other dimensions of request - IOPs, etc
+		if val, ok := claim.Spec.Resources.Requests[kapi.ResourceStorage]; ok {
+			return fmt.Sprintf("waiting for %sB allocation", val.String())
+		}
+		return "waiting to allocate"
+	}
+	// TODO: check for other dimensions of capacity?
+	if val, ok := claim.Status.Capacity[kapi.ResourceStorage]; ok {
+		return fmt.Sprintf("allocated %sB", val.String())
+	}
+	return "allocated unknown size"
+}
+
+func describeVolumeSource(source *kapi.VolumeSource) string {
```

Comment:
I opened an upstream issue to track.

On Aug 11, 2015, at 2:16 PM, Fabiano Franz <notifications@github.com> wrote:

In pkg/cmd/cli/cmd/volume.go
<https://github.com/openshift/origin/pull/4055#discussion_r36778937>:

> +func describePersistentVolumeClaim(claim *kapi.PersistentVolumeClaim) string {
> +	if len(claim.Spec.VolumeName) == 0 {
> +		// TODO: check for other dimensions of request - IOPs, etc
> +		if val, ok := claim.Spec.Resources.Requests[kapi.ResourceStorage]; ok {
> +			return fmt.Sprintf("waiting for %sB allocation", val.String())
> +		}
> +		return "waiting to allocate"
> +	}
> +	// TODO: check for other dimensions of capacity?
> +	if val, ok := claim.Status.Capacity[kapi.ResourceStorage]; ok {
> +		return fmt.Sprintf("allocated %sB", val.String())
> +	}
> +	return "allocated unknown size"
> +}
> +
> +func describeVolumeSource(source *kapi.VolumeSource) string {

Nice, we should later consider moving these upstream.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4055/files#r36778937>.